src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1055,7 +1049,7 @@ private void SetCapacity(int capacity)
                 }
             }
 
-            _count = capacity;
+            _count = count;
             _freeCount = 0;
```

Comment:
`_freeCount = capacity - count;` ?

Reply:
no
- _count is the number of "used" array slots (both free and actual hashset content)
- _freeCount is the number of free slots within the first _count array slots

so the slots starting from index _count to the end of the backing array are not counted in _freeCount
this is beneficial because you can skip this section when iterating the hashset, so you don't want to include them in the free count